main.go

```diff
@@ -358,6 +358,14 @@ func main() {
 		_ = registerClient.Unregister(context.Background(), conf)
 	}()
 
+	// fetch the kubeslice-dns.kubeslice-system service's ClusterIP
+	var clusterIPOfDNSService string
+	if os.Getenv("DNS_SERVICE_NAME") != "" {
```

Comment:
Please move this code to createInitContainerPatch func. Lets avoid changing the func signatures of Review() and createInitContainerPatch().

Reply:
we require the "registerClient" to call kube-apiserver , this client is not available in `createInitContainerPatch()` @bharath-avesha